os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -1970,21 +1976,14 @@ static uint32_t rtk_i2s_resume(uint32_t expected_idle_time, void *param)
 	(void)param;
 
 	/* For PG Sleep, I2S HW will be lost power, thus a reinitialization is required here */
-#ifdef CONFIG_AMEBASMART_I2S2
-	amebasmart_i2s_resume(I2S_NUM_2);
-#endif
-#ifdef CONFIG_AMEBASMART_I2S3
-	amebasmart_i2s_resume(I2S_NUM_3);
-#endif
+	amebasmart_i2s_resume((i2s_port_t)param);
 
 	return 1;
 }
-#endif
 
-#ifdef CONFIG_PM
-void i2s_pminitialize(void)
+static void i2s_pminitialize(uint16_t port)
 {
 	bsp_pm_domain_register("I2S", BSP_I2S_DRV);
-	pmu_register_sleep_callback(PMU_I2S_DEVICE, (PSM_HOOK_FUN)rtk_i2s_suspend, NULL, (PSM_HOOK_FUN)rtk_i2s_resume, NULL);
+	pmu_register_sleep_callback(PMU_I2S_DEVICE, (PSM_HOOK_FUN)rtk_i2s_suspend, (void *)port, (PSM_HOOK_FUN)rtk_i2s_resume, (void *)port);
```

Comment:
``` c
void pmu_register_sleep_callback(u32 nDeviceId, PSM_HOOK_FUN sleep_hook_fun, void *sleep_param_ptr,
								 PSM_HOOK_FUN wakeup_hook_fun, void *wakeup_param_ptr)
{
	PSM_DD_HOOK_INFO *pPsmDdHookInfo = NULL;

	assert_param(nDeviceId < PMU_MAX);
	assert_param((sleep_hook_fun != NULL) || (wakeup_hook_fun != NULL));

	pPsmDdHookInfo = &gPsmDdHookInfo[nDeviceId];
	pPsmDdHookInfo->nDeviceId			= nDeviceId;
	pPsmDdHookInfo->sleep_hook_fun		= sleep_hook_fun;
	pPsmDdHookInfo->sleep_param_ptr		= sleep_param_ptr;
	pPsmDdHookInfo->wakeup_hook_fun		= wakeup_hook_fun;
	pPsmDdHookInfo->wakeup_param_ptr	= wakeup_param_ptr;
}
```

the pmu_register_sleep_callback has static array map for registering device.
so, if you put in port I2S2 and I2S3 using param, the port is overwriten.